### ISSUE TITLE
[contracts] Raid tx error

### ISSUE THREAD

**Issue by cwastche** [BibliothecaDAO/eternum#1394]
To reproduce:
1. Realm A w/ defender army
2. Realm B attacks realm A
3. Realm B army tries to raid

Tx went through on 2nd raid attempt

```Uncaught (in promise) Error: Transaction failed with reason: Transaction execution has failed:
0: Error in the called contract (contract address: 0x03256bd286aff3d2edc85513b69865526ee90f297ac27bbb726e381d854437a0, class hash: 0x05400e90f7e0ae78bd02c77cd75527280470e2fe19c54970dd79dc37a9d3645c, selector: 0x015d40a3d6ca2ac30f4031e42be28da9b056fef9bb7357ac5e85627ee876e5ad):
Error at pc=0:4573:
Cairo traceback (most recent call last):
Unknown location (pc=0:67)
Unknown location (pc=0:1835)
Unknown location (pc=0:2478)
Unknown location (pc=0:3255)
Unknown location (pc=0:3809)
Unknown location (pc=0:3795)

1: Error in the called contract (contract address: 0x01c3310b831c666aeeefbe9ec95fe3140cb32ecb1a279b7898782b0e450c79a8, class hash: 0x03ffe895aab1e77ab9772da474ac6a0ed785c9226973c9d8b623962682c96e1d, selector: 0x018f505bd039d3fbc20655d8359bcda825bd8fac64a17ead4b26d26917199428):
Execution failed. Failure reason: "resource locked for entity 39".

    at ee.waitForTransactionWithCheck (index.ts:72:56)
    at async ee.executeAndCheckTransaction (index.ts:58:27)
    at async ee.battle_leave_and_pillage (index.ts:593:26)
    at async Object.battle_leave_and_pillage (createSystemCalls.ts:260:5)
    at async BattleManager.pillageStructure (BattleManager.ts:271:7)
    at async handleRaid (BattleActions.tsx:83:5)```

**Comment by cwastche** [comment]
Seems like tx fails on successful raids, and not on failures

**Comment by edisontim** [comment]
We've discussed this with @credence0x, it might already be closed by #1385
